lib/cadet_web/controllers/auth_controller.ex

```diff
@@ -6,35 +6,59 @@ defmodule CadetWeb.AuthController do
   use PhoenixSwagger
 
   alias Cadet.Accounts
+  alias Cadet.Accounts.Form.Login
+  alias Cadet.Accounts.Ivle
   alias Cadet.Auth.Guardian
-  alias Cadet.Accounts.Form
 
+  @doc """
+  Receives a /login request with valid attributes (`Login form`).
+
+  If the user is already registered in our database, simply return `Tokens`. If
+  the user has not been registered before, register the user, then return the
+  `Tokens`.
+  """
   def create(conn, %{"login" => attrs}) do
-    changeset = Form.Login.changeset(%Form.Login{}, attrs)
+    changeset = Login.changeset(%Login{}, attrs)
 
     if changeset.valid? do
       login = apply_changes(changeset)
 
-      case Accounts.sign_in(login.email, login.password) do
-        {:ok, user} ->
-          {:ok, access_token, _} =
-            Guardian.encode_and_sign(user, %{}, token_type: "access", ttl: {1, :hour})
-
-          {:ok, refresh_token, _} =
-            Guardian.encode_and_sign(user, %{}, token_type: "refresh", ttl: {52, :weeks})
-
-          render(conn, "token.json", access_token: access_token, refresh_token: refresh_token)
+      case Ivle.fetch_nusnet_id(login.ivle_token) do
```

Comment:
`with` construct is better here to prevent pyramid of doom https://hexdocs.pm/elixir/Kernel.SpecialForms.html#with/1

Reply:
Done with b2834fc.